cmd/sdkstats/main.go

```diff
@@ -7,46 +7,188 @@ import (
 	"fmt"
 	"io/ioutil"
 	"log"
+	"os"
+	"regexp"
+	"strings"
 
-	"github.com/go-github-metrics-1/pkg/sdkstats"
+	"github.com/learnoperators/go-github-metrics/pkg/sdkstats"
 
 	"github.com/google/go-github/github"
 	"golang.org/x/oauth2"
 )
 
+type helmVersionParser struct{}
+type baseVersionParser struct {
+	searchQ      string
+	searchLatest string
+}
+type gomodVersionParser struct {
+	searchQ string
+}
+type unknownVersionParser struct{}
+
 func main() {
 	var token string
 	flag.StringVar(&token, "token", "", "GitHub API token")
 	flag.Parse()
-	args := flag.Args()
+	if len(token) == 0 {
+		log.Fatal("GITHUB API TOKEN MUST BE ENTERED \n Usage: './main --token=YOURTOKEN'")
+	}
 	ctx := context.Background()
 	ts := oauth2.StaticTokenSource(
-		&oauth2.Token{AccessToken: args[0]},
+		&oauth2.Token{AccessToken: token},
 	)
 	tc := oauth2.NewClient(ctx, ts)
-	client := github.NewClient(tc)
+	client := sdkstats.Client{Client: github.NewClient(tc)}
 
-	var queries []string
-
-	queries = append(queries, "filename:Dockerfile quay.io/operator-framework/helm-operator")
-	queries = append(queries, "filename:Dockerfile quay.io/operator-framework/ansible-operator")
-	queries = append(queries, "filename:go.mod github.com/operator-framework/operator-sdk")
-	queries = append(queries, "filename:Gopkg.toml github.com/operator-framework/operator-sdk")
+	queries := []sdkstats.RepoMetadataQuery{
+		/*sdkstats.RepoMetadataQuery{
+			ProjectType: "helm",
+			Queries:     []string{"filename:Dockerfile quay.io/operator-framework/helm-operator"},
+			VersionParser: &baseVersionParser{
+				searchQ: "quay.io/operator-framework/helm-operator",
+			},
+		},
+		sdkstats.RepoMetadataQuery{
+			ProjectType: "ansible",
+			Queries:     []string{"filename:Dockerfile quay.io/operator-framework/ansible-operator"},
+			VersionParser: &baseVersionParser{
+				searchQ: "quay.io/operator-framework/ansible-operator",
+			},
+		},*/
+		sdkstats.RepoMetadataQuery{
+			ProjectType: "go.mod",
+			Queries: []string{
+				"filename:go.mod github.com/operator-framework/operator-sdk",
```

Comment:
Why we are not using `org:` and `repo:` and entity as filter`?
See here for an example: https://developer.github.com/v3/pulls/

If we would like to have all pr's that were merged : 
<img width="758" alt="Screenshot 2019-11-14 at 13 12 29" src="https://user-images.githubusercontent.com/7708031/68859981-6f6bd400-06e0-11ea-8d54-96774a9bd7de.png">

Also, we can aggregate filters, as for example a range of data.